.github/workflows/build.yml

```diff
@@ -0,0 +1,24 @@
+name: Java CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
```

Comment:
to confirm, will this build on 11? If so, shouldn't we also run on 11 (probably with a matrix strategy)

Reply:
This will build with jdk17 targeting java 11, minimum current eclipse requirement.
Building with java 11 might be more correct (maven observes that dependency resolution might depend on JDK version).
But the plugin seems ok (and manual class file inspection shows "version 55.0 (Java SE 11)").

Reply:
The built contents hopefully arent different if we're targeting an older Java version, but the tests might expose that something works in one Java version but not another.